src/Microsoft.SqlTools.ServiceLayer/ModelManagement/ModelOperations.cs

```diff
@@ -0,0 +1,416 @@
+				{
+					command.CommandText = GetConfigTableVerificationQuery(request.DatabaseName, request.TableName, request.SchemaName);
+
+					command.ExecuteNonQuery();
+					using (IDataReader reader = command.ExecuteReader())
+					{
+						while (reader.Read())
+						{
+							return reader.GetInt32(0);
+						}
+					}
+					return 0;
+				}
+			});
+
+			return result == 1;
+		}
+
+		private TResult WithDbChange<T, TResult>(IDbConnection connection, T request, Func<T, TResult> operation) where T : ModelRequestBase
+		{
+			string currentDb = connection.Database;
+			if (connection.Database != request.DatabaseName)
+			{
+				connection.ChangeDatabase(request.DatabaseName);
+			}
+			TResult result = operation(request);
+
+			if (connection.Database != currentDb)
+			{
+				connection.ChangeDatabase(currentDb);
+			}
+			return result;
+		}
+
+		private ModelMetadata LoadModelMetadata(IDataReader reader)
+		{
+			return new ModelMetadata
+			{
+				Id = reader.GetInt32(0),
+				ModelName = reader.GetString(1),
+				Description = reader.IsDBNull(2) ? string.Empty : reader.GetString(2),
+				Version = reader.IsDBNull(3) ? string.Empty : reader.GetString(3),
+				Created = reader.IsDBNull(4) ? string.Empty : reader.GetDateTime(4).ToString(),
+				Framework = reader.IsDBNull(5) ? string.Empty : reader.GetString(5),
+				FrameworkVersion = reader.IsDBNull(6) ? string.Empty : reader.GetString(6),
+				DeploymentTime = reader.IsDBNull(7) ? string.Empty : reader.GetDateTime(7).ToString(),
+				DeployedBy = reader.IsDBNull(8) ? string.Empty : reader.GetString(8),
+				RunId = reader.IsDBNull(9) ? string.Empty : reader.GetString(9),
+				ContentLength = reader.GetInt64(10),
+			};
+		}
+
+		private const string ModelSelectColumns = @"
+        SELECT model_id, model_name, model_description, model_version, model_creation_time, model_framework, model_framework_version, model_deployment_time, User_Name(deployed_by), run_id, 
+        len(model)";
+
+		private static string GetThreePartsTableName(string dbName, string tableName, string schemaName)
+		{
+			return $"[{CUtils.EscapeStringCBracket(dbName)}].[{CUtils.EscapeStringCBracket(schemaName)}].[{CUtils.EscapeStringCBracket(tableName)}]";
+		}
+
+		private static string GetTwoPartsTableName(string tableName, string schemaName)
+		{
+			return $"[{CUtils.EscapeStringCBracket(schemaName)}].[{CUtils.EscapeStringCBracket(tableName)}]";
+		}
+
+		private static string GetSelectModelsQuery(string dbName, string tableName, string schemaName)
+		{
+			return $@"
+        {ModelSelectColumns}
+        FROM {GetThreePartsTableName(dbName, tableName, schemaName)}
+		WHERE model_name not like 'MLmodel' and model_name not like 'conda.yaml'
+		ORDER BY model_id";
+		}
+
+		private static string GetConfigTableVerificationQuery(string dbName, string tableName, string schemaName)
+		{
+			string twoPartsTableName = GetTwoPartsTableName(CUtils.EscapeStringSQuote(tableName), CUtils.EscapeStringSQuote(schemaName));
```

Comment:
I don't think escaping the single quote here is correct - you're already bracketing it and so this will just unnecessarily escape single quotes (which would likely break the script)

Reply:
hmm I thought so to but my test failed when I only escaped with bracket.

Reply:
What "test"?

What's the final query look like when you have a table with a ' in its name?

Reply:
this test I added to ModelOperationsTests in this PR.
 [Test]
        public void VerifyImportTableShouldReturnTrueGivenTableNameThatNeedsEscaping()
        {
            bool expected = true;
            bool actual = VerifyModelOperation((dbConnection, databaseName , tableName) =>
            {
                ModelOperations modelOperations = new ModelOperations();
                ModelRequestBase request = new ModelRequestBase
                {
                    DatabaseName = databaseName,
                    SchemaName = "dbo",
                    TableName = tableName
                };
                modelOperations.ConfigureImportTable(dbConnection, request);
                return modelOperations.VerifyImportTable(dbConnection, request);
            }, null, "models[]'");

            Assert.AreEqual(expected, actual);
        }

---
In reply to: [542915398](https://github.com/microsoft/sqltoolsservice/pull/1138#discussion_r542915398) [](ancestors = 542915398)

Reply:
this line was failing: IF EXISTS (SELECT * FROM syscolumns WHERE ID=OBJECT_ID('[dbo].[models[]]') AND NAME='model_name')

---
In reply to: [542916949](https://github.com/microsoft/sqltoolsservice/pull/1138#discussion_r542916949) [](ancestors = 542916949,542915398)

Reply:
Ah, yeah I didn't see that it was also being single quote escaped. Yeah that looks fine then